cvmfs/sync_mediator.cc

```diff
@@ -807,7 +821,34 @@ void SyncMediator::AddFile(SharedPtr<SyncItem> entry) {
     file_queue_[entry->GetUnionPath()] = entry;
     pthread_mutex_unlock(&lock_file_queue_);
     // Spool the file
-    params_->spooler->Process(entry->CreateIngestionSource());
+    params_->spooler->Process(SyncItem);
+  }
+
+  // publish statistics counting
+  if (counters_.IsValid()) {
+    Inc(counters_->n_files_added);
+
+    uint64_t size = 0;
+    if (GetFileSize(entry->GetScratchPath()) == -1) {
+      SyncItem->GetSize(&size);
+    } else {
+      size = GetFileSize(entry->GetScratchPath());
+    }
+
+    Xadd(counters_->n_bytes_added, size);
```

Comment:
Perhaps this can go down to the `else` branch, to avoid adding only to subtract again if the file has been changed and not added.